### PR TITLE
UX fixed for date field

### PR DESCRIPTION
**Pull request by Shruti-Apte**
### **What :**
3rd and 4th task from following ticket:
(https://www.notion.so/saeloun/Invoice-page-UX-feedback-c94100068cbf41708986ccccfdb92f08)

- By clicking anywhere  on date field date picker is displayed.  
- By clicking outside, it is closing.
- UI fixed for invoice number and reference number.
### **Why :**
- UX requirement.
(Earlier only on clicking pencil icon we were able to edit the date.)

### **Preview :**
**Code:**

```
<div
          className="hoverPencil cursor-pointer"
          onClick={() => setShowDueDatePicker(!showDueDatePicker)}
        >
          <p className="mt-4 flex text-xs font-normal text-miru-dark-purple-1000">
            <span>Due Date</span>
            <button className="invisible ml-2">
              <EditIcon color="#1D1A31" size={13} />
            </button>
          </p>
          {showDueDatePicker && (
            <CustomDatePicker
              date={dueDate}
              handleChange={handleDueDatePicker}
              setVisibility={setShowDueDatePicker}
            />
          )}
          <p className="text-base font-normal text-miru-dark-purple-1000">
            {getDueDate}
          </p>
        </div>
```

**Screenshots:**
Screenshot 2023-02-07 at 1 36 33 PM

**Video:**

https://www.loom.com/share/9ece6ca38fab487ba3b39d4b0c2e724c
 Updated video : https://www.loom.com/share/143bc92321874ca39536d5ef227cbbe9

### **Todos** (testing):
- The user should be able to click on the date field itself to edit the date .
- A custom calendar should open and user should be able to select another year from the dropdown , select a date.
- After selecting a date the calendar should close and the date field should not be in edit UI mode.
- If the user opens calendar, it should close as soon as the user click anywhere outside the calendar.
- User should be able to add references number add invoice number when clicking on the placeholder.

### CODE REVIEW

**Comment by supriya3105** [comment]
@Shruti-Apte 
1. Whenever the calendar is open, the date field should be visible to the user. 
2 Also the calendar should open exactly below the date field. I think right now its open on the right side.

**Comment by apoorv1316** [comment]
@Shruti-Apte Please work on Supriya's comments and I will approve it then.

**Comment by supriya3105** [comment]
> @Shruti-Apte
> 
> 1. Whenever the calendar is open, the date field should be visible to the user.
>    2 Also the calendar should open exactly below the date field. I think right now its open on the right side.

For now keeping the calendar alignment to the right i.e. as it is now. The first part needs to be fixed.

File: app/javascript/src/common/CustomDatePicker/index.tsx [hunk 1]
```diff
@@ -1,11 +1,23 @@
-import React from "react";
+import React, { useRef } from "react";
 
 import { getMonth, getYear } from "date-fns";
 import { CaretCircleLeftIcon, CaretCircleRightIcon } from "miruIcons";
 import DatePicker from "react-datepicker";
 import "react-datepicker/dist/react-datepicker.css";
 
-const CustomDatePicker = ({ handleChange, date }) => {
+import { useOutsideClick } from "../../helpers/outsideClick";
```
Comment (prasanthchaduvula) [inline app/javascript/src/common/CustomDatePicker/index.tsx:8]: Here please change the import to `import { useOutsideClick } from "helpers";`

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Comment by Shruti-Apte** [comment]
Also worked on 3rd task from the ticket as it was a very small change. 
listing out the changes (discussed with supriya on call): 
1. Order of the fields is changed. (date section is 3rd now ).
2. Added placeholder for invoice number and reference number.
3. Client selection dropdown issue fixed (pointed out by sonam on call).
cc: @supriya3105 @apoorv1316 
Updating the PR description as well.

**Review by prasanthchaduvula — APPROVED** [review]

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@Shruti-Apte Feedback for the PR https://www.loom.com/share/ee50d0d98feb40b6820380624701d424.

**Comment by Shruti-Apte** [comment]
> @Shruti-Apte Feedback for the PR https://www.loom.com/share/ee50d0d98feb40b6820380624701d424.

Connected with akash for review, he is updating the designs. Will work on it.

**Comment by Shruti-Apte** [comment]
> @Shruti-Apte Feedback for the PR https://www.loom.com/share/ee50d0d98feb40b6820380624701d424.
@apoorv1316 Updated the design and UX according to the discussion with akash and supriya! can you review it!
You can find the design specs here https://www.notion.so/saeloun/Invoice-page-UX-feedback-c94100068cbf41708986ccccfdb92f08

**Comment by apoorv1316** [comment]
@Shruti-Apte Please watch this loom video https://www.loom.com/share/75b79d5ce0d94fbb9793ea1ccbb63c2d

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**